### PR TITLE
D.O. Rework: Topological PP

### PR DESCRIPTION
**Pull request by PaulTalbot-INL**
Got the topological postprocessor working for the new data objects.

### CODE REVIEW

**Comment by PaulTalbot-INL** [comment]
Any comments on this PR?  This includes addExpectedMeta for the PostProcessor base class, so it would be nice to have in.

File: framework/PostProcessors/TopologicalDecomposition.py [hunk 1]
```diff
@@ -23,6 +23,7 @@
 #External Modules------------------------------------------------------------------------------------
 import numpy as np
 import time
+import itertools
```
Comment (alfoa) [inline framework/PostProcessors/TopologicalDecomposition.py]: you are not using this. right?

File: framework/PostProcessors/TopologicalDecomposition.py [hunk 2]
```diff
@@ -116,47 +117,27 @@ def inputToInternal(self, currentInp):
...
-    if 'SamplerType' in inputDict['metadata'].keys():
-      pass
+                        ' postprocessor only accepts PointSet DataObjects for input. ',
+                        ' Requested: ', type(currentInp))
+    # now we know we have a PointSet
+    ## TODO FIXME maintaining old structure for now, in the future convert to use DataObject directly
+    ##    and not bother with inputToInternal
+    ##    This works particularly well since we only accept point sets.
+    data = currentInp.asDataset(outType='dict')['data']
+    inputDict = {'features':dict((var,np.array(data[var],dtype=float)) for var in self.parameters['features']),
```
Comment (alfoa) [inline framework/PostProcessors/TopologicalDecomposition.py]: you should not need the casting here anymore right?

File: framework/PostProcessors/TopologicalDecomposition.py [hunk 3]
```diff
@@ -318,7 +315,7 @@ def run(self, inputIn):
       self.inputData[:, i] = myDataIn[lbl.encode('UTF-8')]
 
     if self.weighted:
-      self.weights = inputIn[0].getMetadata('PointProbability')
+      self.weights = np.array(internalInput['metadata']['PointProbability'],dtype=float)
```
Comment (alfoa) [inline framework/PostProcessors/TopologicalDecomposition.py]: same as above

File: framework/PostProcessors/TopologicalDecomposition.py (hunk 1 above)
Comment (PaulTalbot-INL) [reply framework/PostProcessors/TopologicalDecomposition.py]: Apparently not.  Removed.

File: framework/PostProcessors/TopologicalDecomposition.py (hunk 2 above)
Comment (PaulTalbot-INL) [reply framework/PostProcessors/TopologicalDecomposition.py]: True!  Fixed.

File: framework/PostProcessors/TopologicalDecomposition.py (hunk 3 above)
Comment (PaulTalbot-INL) [reply framework/PostProcessors/TopologicalDecomposition.py]: Done.

**Pull request merged**